### ISSUE TITLE
[@typescript-eslint/no-unused-vars] ParameterDecorator mark unused when i use default value

### ISSUE THREAD

**Issue by DiamondYuan** [typescript-eslint/typescript-eslint#1292]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

```TS
        import { ParameterDecoratorFactory } from 'decorators';
        export class Qux {
           bar(@ParameterDecoratorFactory(true) baz: number = 0) {
               console.log(baz);
           }
        }
```

**Expected Result**

no error

**Actual Result**

error

**Additional Info**

**Versions**

latest

**Comment by DiamondYuan** [comment]
![image](https://user-images.githubusercontent.com/9692408/69938545-93207e00-1518-11ea-9f47-2128b53ffbb5.png)

**Comment by bradzacher** [comment]
Please use `no-unused-vars-experimental` instead.
